Horizon.Database/DatabaseChecker.cs

```diff
@@ -0,0 +1,315 @@
+        {
+            Console.WriteLine("Checking if we can login to database with master table ...");
+            try
+            {
+                context.Database.OpenConnection();
+                Console.WriteLine("Initial master connection successful.");
+                context.Database.CloseConnection();
+                return true;
+            }
+            catch
+            {        
+                Console.WriteLine("Unable to connect to master table.");
+                return false;
+            }
+        }
+
+        private bool IsServerInitialized()
+        {
+            var databaseName = context.Database.GetDbConnection().Database;
+            string sql = $"SELECT COUNT(*) FROM sys.databases WHERE name = '{horizonDatabase}'";
+            return QueryDatabaseInt(sql) > 0;
+        }
+
+
+        public void WaitForDatabase(int pollingIntervalSeconds = 10)
+        {
+
+            while(!IsServerAlive()) {
+                Console.WriteLine($"Waiting {pollingIntervalSeconds} seconds before polling again ...");
+                Thread.Sleep(pollingIntervalSeconds * 1000);
+            }
+
+            if (!IsServerInitialized()) {
+                // Initialize database
+                Console.WriteLine($"Need to initialize database!");
+                
+                // Run CREATE DATABASE script
+                Console.WriteLine($"Running CREATE_DATABASE.sql ...");
+                ExecuteSqlScript(Path.Combine(folderPath, "scripts", "CREATE_DATABASE.sql"));
+
+                // Run CREATE TABLE script
+                Console.WriteLine($"Running CREATE_TABLES.sql ...");
+                ExecuteSqlScript(Path.Combine(folderPath, "scripts", "CREATE_TABLES.sql"));
+
+                // Create admin middleware user
+                CreateAdminUser();
+
+                // Set all app settings into the database
+                SetAppSettings();
+            } else {
+                Console.WriteLine($"Database already initialized!");
+            }
+
+        }
+
+        public void CreateAdminUser() {
+            Console.WriteLine("Creating admin middleware user ...");
+
+            string createAdmin = $@"
+                INSERT INTO accounts.account (account_name, account_password, create_dt, is_active, app_id, reset_password_on_next_login)
+                VALUES ('{middlewareAdminUser}', '{middlewareAdminPassword}', getdate(), 1, 0, 0);
+            ";
+            ExecuteSqlCommand(createAdmin);
+
+            // Add database role 
+            int roleId = QueryDatabaseInt($"SELECT role_id FROM {horizonDatabase}.KEYS.roles where role_name = 'database'");
+            int accountId = QueryDatabaseInt($"SELECT account_id FROM {horizonDatabase}.accounts.account where account_name = '{middlewareAdminUser}'");
+            ExecuteSqlCommand($"INSERT INTO {horizonDatabase}.accounts.user_role VALUES({accountId}, {roleId}, GETDATE(), GETDATE(), null)");
+        }
+
+        public void SetAppSettings() {
```

Comment:
Maybe rename to InitializeAppSettings or SetInitialAppSettings

SetAppSettings by itself would make me think its a normal function for setting any setting